demos/classification_demo/main.cpp

```diff
@@ -0,0 +1,537 @@
+    double scale = static_cast<double>(modelInputResolution) / std::min(image.cols, image.rows);
+
+    cv::Mat resizedImage;
+    cv::resize(image, resizedImage, cv::Size(), scale, scale);
+
+    cv::Rect imgROI;
+    if (resizedImage.cols >= resizedImage.rows) {
+        int fromWidth = resizedImage.cols/2 - modelInputResolution/2;
+        imgROI = cv::Rect(fromWidth, 0, modelInputResolution, modelInputResolution);
+    } else {
+        int fromHeight = resizedImage.rows/2 - modelInputResolution/2;
+        imgROI = cv::Rect(0, fromHeight, modelInputResolution, modelInputResolution);
+    }
+
+    return resizedImage(imgROI);
+}
+
+std::vector<std::vector<unsigned>> topResults(Blob& inputBlob, unsigned numTop) {
+    using currentBlobType = PrecisionTrait<Precision::FP32>::value_type;
+    TBlob<currentBlobType>& tblob = dynamic_cast<TBlob<currentBlobType>&>(inputBlob);
+    size_t batchSize =  tblob.getTensorDesc().getDims()[0];
+    numTop = static_cast<unsigned>(std::min<size_t>(size_t(numTop), tblob.size()));
+    
+    std::vector<std::vector<unsigned>> output(batchSize);
+    for (size_t i = 0; i < batchSize; i++) {
+        size_t offset = i * (tblob.size() / batchSize);
+        currentBlobType *batchData = tblob.data() + offset;
+        std::vector<unsigned> indices(tblob.size() / batchSize);
+        std::iota(std::begin(indices), std::end(indices), 0);
+        std::partial_sort(std::begin(indices), std::begin(indices) + numTop, std::end(indices),
+                          [&batchData](unsigned l, unsigned r) {
+                              return batchData[l] > batchData[r];
+                          });
+
+        indices.resize(numTop);
+        output[i].assign(indices.begin(), indices.begin() + numTop);
+    }
+    
+    return output;
+}
+
+int main(int argc, char *argv[]) {
+    try {
+        if (!ParseAndCheckCommandLine(argc, argv)) {
+            return 0;
+        }
+
+        // -----------------------------------------Read input images-----------------------------------------
+        std::vector<std::string> imageNames;
+        std::vector<cv::Mat> inputImages;
+        parseInputFilesArguments(imageNames);
+        if (imageNames.empty()) throw std::runtime_error("No images provided");
+        std::sort(imageNames.begin(), imageNames.end());
+        for (size_t i = 0; i < imageNames.size(); i++) {
+            const std::string& name = imageNames[i];
+            const cv::Mat& tmpImage = cv::imread(name);
+            if (tmpImage.data == nullptr) {
+                std::cerr << "Could not read image " << name << '\n';
+                imageNames.erase(imageNames.begin() + i);
+                i--;
+            } else {
+                inputImages.push_back(tmpImage);
+                size_t lastSlashIdx = name.rfind('/');
+                if (lastSlashIdx == std::string::npos) {
+                    lastSlashIdx = name.rfind('\\'); // for compatibility with Windows
+                }
```

Comment:
This isn't correct either, because on Windows a path can also have forward slashes in it. You should just use `find_last_of`.